doc/mpls/MPLS_hld.md

```diff
@@ -0,0 +1,509 @@
+      - [Configuration and Management requirements](#configuration-and-management-requirements)
+      - [Scalability Requirements](#scalability-requirements)
+      - [Warm Boot Requirements](#warm-boot-requirements)
+      - [Future Requirements](#future-requirements)
+    - [Architecture Design](#architecture-design)
+    - [High-Level Design](#high-level-design)
+      - [Overview](#overview-1)
+      - [Database Changes](#database-changes)
+        - [APPL DB](#appl-db)
+          - [INTERFACE_TABLE](#interface_table)
+          - [ROUTE_TABLE](#route_table)
+          - [LABEL_ROUTE_TABLE](#label_route_table)
+      - [Software Modules](#software-modules)
+        - [NetLink](#netlink)
+          - [Functions](#functions)
+        - [IntfMgr](#intfmgr)
+        - [FPM Syncd](#fpm-syncd)
+          - [Functions](#functions-1)
+        - [IntfsOrch](#intfsorch)
+          - [Functions](#functions-2)
+        - [RouteOrch](#routeorch)
+          - [Functions](#functions-3)
+        - [NeighOrch](#neighorch)
+          - [Functions](#functions-4)
+        - [Label/LabelStack](#labellabelstack)
+        - [NextHopKey](#nexthopkey)
+        - [Syncd](#syncd)
+    - [SAI API](#sai-api)
+      - [Router Interface](#router-interface)
+      - [MPLS](#mpls)
+      - [Next Hop](#next-hop)
+      - [Next Hop Group](#next-hop-group)
+    - [Configuration and management](#configuration-and-management)
+      - [CLI Enhancements](#cli-enhancements)
+      - [Config DB Enhancements](#config-db-enhancements)
+        - [INTERFACE](#interface)
+        - [PORTCHANNEL_INTERFACE](#portchannel_interface)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+    - [Open/Action items - if any](#openaction-items---if-any)
+### Revision
+|  Rev  | Date  |  Author  | Change Description |
+| :---: | :---: | :------: | ------------------ |
+|  0.1  |       | A Pokora | Initial version    |
+
+### Scope
+
+This document provides general information about the MPLS feature implementation in SONiC.
+
+### Definitions/Abbreviations
+| Abbreviation | Description                           |
+| ------------ | ------------------------------------- |
+| cRPD         | Containerized Routing Protocol Daemon |
+| ECMP         | Equal Cost Multi-Path                 |
+| MPLS         | Multi-Label Packet Switching          |
+
+### Overview
+
+This document provides general information about the MPLS feature implementation in SONiC.
+
+### Requirements
+
+This section describes the SONiC requirements for:
+- MPLS routes feature.
+- Weighted ECMP routes feature.
+
+#### Functional requirements
+- Support to enable/disable MPLS per Interface.
+- Support for MPLS Push, Pop, and Swap routes.
```

Comment:
PHP support is needed.

Reply:
can we push PHP support in follow-on phase? It probably need its own HLD. As well as other features (e.g. ttl/exp mode..uniform vs. pipe)

Reply:
In the mpls subgroup, this was raised as a major gap missing, as it's a common function widely used today.

Reply:
Yes, we will update HLD, PHP will be supported by treating implicit NULL to pop and swap for explicit NULL label.